### PR TITLE
Branch appointment storage

### PR DESCRIPTION
**Pull request by officialchengyud**

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        return jsonAdaptedAppt != null
+                && jsonAdaptedAppt.doctorNric.equals(this.doctorNric)
+                && jsonAdaptedAppt.patientNric.equals(this.patientNric)
+                && jsonAdaptedAppt.appointmentDate.equals(this.appointmentDate)
+                && jsonAdaptedAppt.appointmentId.equals(this.appointmentId);
+
+    }
+
+}
```
Comment (alfaloo) [inline src/main/java/seedu/address/storage/JsonAdaptedAppointment.java:123]: this is good stuff

File: src/test/java/seedu/address/storage/JsonAdaptedAppointmentTest.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+     * The method should return false as the objects are of different types.
+     *
+     * @throws Exception if an error occurs during the test.
+     */
+    @Test
+    public void toJsonAdapatedAppointment_differentObjectType_returnsFalse() throws Exception {
+        JsonAdaptedAppointment jsonAdaptedAppointment = new JsonAdaptedAppointment(APPOINTMENT_2);
+        assertFalse(jsonAdaptedAppointment.equals(APPOINTMENT_2));
+    }
+}
```
Comment (alfaloo) [inline src/test/java/seedu/address/storage/JsonAdaptedAppointmentTest.java:210]: I like the detailed tests Chengyu!

**Review by alfaloo — APPROVED** [review]

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 3]
```diff
@@ -47,13 +54,24 @@ public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
...
+        // Add appointment models to the address book
+        for (JsonAdaptedAppointment jsonAdaptedAppointment : appointments) {
+            Appointment appointment = jsonAdaptedAppointment.toModelType();
+            if (addressBook.hasAppointment(appointment)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_APPOINTMENTS);
+            }
+            addressBook.addAppointment(appointment);
+        }
         return addressBook;
     }
```
Comment (Kappaccinoh) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:77]: nice integration of appointment into existing person storage

File: src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java [hunk 4]
```diff
@@ -36,12 +53,20 @@ public void toModelType_invalidPersonFile_throwsIllegalValueException() throws E
...
+                    dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_duplicateAppointments_throwsIllegalValueException() throws Exception {
+        JsonSerializableAddressBook dataFromFile = JsonUtil.readJsonFile(DUPLICATE_APPOINTMENT_FILE,
+                JsonSerializableAddressBook.class).get();
+        assertThrows(IllegalValueException.class, JsonSerializableAddressBook.MESSAGE_DUPLICATE_APPOINTMENTS,
+                dataFromFile::toModelType);
+    }
```
Comment (Kappaccinoh) [inline src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java:70]: great job including the tests that we left out previously - our code coverage is looking healthy

**Review by Kappaccinoh — APPROVED** [review]

**Pull request merged**